src/view/views/modal/SettingsWindow.tsx

```diff
@@ -0,0 +1,88 @@
+import { useState, useEffect } from 'react';
+import {
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  ModalCloseButton,
+  Button,
+  Flex
+} from '@chakra-ui/react';
+import useSetting from './use-settings.ts';
+import { SceneConfig } from '../../../scene/types/scene-config.ts';
+
+interface SettingsWindowProps {
+  isOpen: boolean;
+  onClose: () => void;
+  sceneConfig: SceneConfig;
+  muteARButton?: boolean;
+}
+
+/**
+ * SettingsWindow component for managing settings.
+ *
+ * @param {object} props - Component props
+ * @param {boolean} props.isOpen - Indicates whether the modal is open.
+ * @param {function} props.onClose - Function to call when closing the modal.
+ * @param {boolean} [props.muteARButton] - Optional flag to control whether the AR button should be disabled.
+ */
+function SettingsWindow({
+  isOpen,
+  onClose,
+  muteARButton
+}: SettingsWindowProps) {
+  const { arEnabled, toggleAR } = useSetting();
+  const [isMusicEnabled, setMusicEnabled] = useState(false);
+  const [arButtonDisabled, setArButtonDisabled] = useState(false);
+
+  useEffect(() => {
+    setArButtonDisabled(!arEnabled);
+  }, [arEnabled]);
```

Comment:
What's the necessity of this? Can you just base button states on `arEnabled`?

Reply:
resolved